From f122ee0620380b7c21edce722cfe347f10c8c827 Mon Sep 17 00:00:00 2001
From: Joshua Lock <joshua.lock@intel.com>
Date: Tue, 03 Sep 2013 10:01:26 +0000
Subject: brasero-app: don't try and show the main window if it hasn't been created

It's possible to open brasero without creating the main window, such as
when opening an image file directly triggering the image burning dialogue.
In these cases trying to show the main window when the dialogue is
dismissed is an error, brasero should just exit.

https://bugzilla.gnome.org/show_bug.cgi?id=655513
---
diff --git a/src/brasero-app.c b/src/brasero-app.c
index d345941..8d10781 100644
--- a/src/brasero-app.c
+++ b/src/brasero-app.c
@@ -2013,6 +2013,9 @@ brasero_app_run_mainwin (BraseroApp *app)
 
 	priv = BRASERO_APP_PRIVATE (app);
 
+	if (!priv->mainwin)
+		return FALSE;
+
 	if (priv->mainwin_running)
 		return TRUE;
 
@@ -2055,6 +2058,8 @@ brasero_app_init (BraseroApp *object)
 
 	priv = BRASERO_APP_PRIVATE (object);
 
+	priv->mainwin = NULL;
+
 	/* Load settings */
 	priv->setting = brasero_setting_get_default ();
 	brasero_setting_load (priv->setting);
--
cgit v0.9.2
